File: cpp/model/Offer.cpp
#include "Offer.h"

Offer::Offer(const SpecialOfferType& offerType, const Product& product, double argument)
        : offerType(offerType), product(product), argument(argument) {}

SpecialOfferType Offer::getOfferType() const {
        return offerType;
}

Product Offer::getProduct() const {
        return product;
}

double Offer::getArgument() const {
        return argument;
}

File: php/src/Model/Offer.php
<?php

declare(strict_types=1);

namespace Supermarket\Model;

class Offer
{
    public function __construct(
        private SpecialOfferType $offerType,
        private Product $product,
        private float $argument
    ) {
    }

    public function getArgument(): float
    {
        return $this->argument;
    }

    public function getProduct(): Product
    {
        return $this->product;
    }

    public function getOfferType(): SpecialOfferType
    {
        return $this->offerType;
    }
}
